cluster-autoscaler/core/scaleup/orchestrator/orchestrator.go

```diff
@@ -449,6 +462,29 @@ func (o *ScaleUpOrchestrator) ComputeExpansionOption(
 		return option
 	}
 
+	autoscalingOptions, err := nodeGroup.GetOptions(o.autoscalingContext.NodeGroupDefaults)
+	if err != nil {
+		klog.Errorf("Failed to get autoscaling options for node group %s: %v", nodeGroup.Id(), err)
+	}
+	if autoscalingOptions != nil && autoscalingOptions.AtomicScaleUp {
+		// If atomic, there should be no similar node groups for balancing
+		// Also the number of pods should be capped according to the node count.
+		option.IsAtomic = true
+		// Limit the number of nodes we estimate to the max scale up possible.
+		currentTargetSize, err := nodeGroup.TargetSize()
+		if err != nil {
+			klog.Errorf("Failed to get node group size: %v", err)
+			return option
+		}
+		nodeEstimationCap := nodeGroup.MaxSize() - currentTargetSize
+		limiter := estimator.NewThresholdBasedEstimationLimiter(nodeEstimationCap, 0)
+		estimator := estimator.NewBinpackingNodeEstimator(o.autoscalingContext.PredicateChecker, o.autoscalingContext.ClusterSnapshot, limiter, estimator.NewDecreasingPodOrderer())
```

Comment:
From looking at the code it's not very clear why you need a different estimator than the regular one, and I think in general it would be better to use the estimator builder from autoscaling context for consistency... AFAICT the only problem with that would be a cap for a node number (MaxNodesPerScaleUp)? Can you add a comment explaining that?

Reply:
Added a comment explaining why we create a different estimator. Maybe, we can also change the definition of `EstimatorBuilder` to also take the limiter instance as an argument?

Reply:
If we want to do it this way, please extend estimator builder. There is a possibility some other change will be introduced, changing this behaviour (moving limiters to processor), but not sure when. Once I know more I will comment here.

Reply:
It turns out we don't actually need to create a new estimator, existing one works fine for this purpose